### ISSUE TITLE
Don't use munge_points_to_quads() to renders sprites when a TexMatrixAttrib is present and a point shader is configured

### ISSUE THREAD

**Issue by el-dee** [panda3d/panda3d#607]
When rendering point sprites with a TexMatrixAttrib applied on the geometry, the code force software point sprites rendering using munge_points_to_quads () even if there is a point shader configured on the node.

On the other hand, when using per point sprite size, if a valid point shader is applied on the node the rendering is done by the hardware using the provided shader :

cullableObject.cxx:
```
      if (unsupported_bits & Geom::GR_per_point_size) {
        // If we have a shader that processes the point size, we can assume it
        // does the right thing.
        const ShaderAttrib *sattr;
        if (_state->get_attrib(sattr) && sattr->get_flag(ShaderAttrib::F_shader_point_size)) {
          unsupported_bits &= ~Geom::GR_per_point_size;
        }
      }
```

I believe there should be a similar check for GR_point_sprite_tex_matrix (unless there are side-effects I'm not aware of)

**Comment by rdb** [comment]
I guess the suggested assumption is that if there is a shader, it is likely to handle texture matrix correctly?  I wonder if we need a separate ShaderAttrib flag for this or if this is a generally safe assumption we can make.

**Comment by el-dee** [comment]
Yes, my understanding is that if one use a custom shader for rendering hardware point sprites, one should handle all the input provided by Panda3D.
Now I don't think we need a separate ShaderAttrib, at least until we have the new shader generator, as for now if you use a custom shader you have to do all the vertex and fragment code anyway, so also handle the TexMatrix, if any.

**Comment by rdb** [comment]
Do you have a need for this change to go into a 1.10 release or do you work with the master branch anyway?

**Comment by el-dee** [comment]
I'm still using 1.10.x (when I eventually release my app I want to rely on a stable version of Panda3D).

On the other hand this is not a critical problem, I discovered it because I had a trivial TexMatrix without transformation, applied on the node. So the workaround was to remove the attrib :)

If needed I can make a PR fixing this, it's quite trivial, but first I wanted to be sure that there are no unforeseen consequences.

**Comment by rdb** [comment]
Sure, please go ahead and make a PR for this against `release/1.10.x`, thanks!

**Comment by el-dee** [comment]
Actually I'm no longer so sure about the fix :  the shader attrib is F_shader_point_**size** : it informs panda that the shader will provide the point size, using that attrib to assume it will also handle the tex matrix attrib is a bit of a stretch.

I can just check that there is a shader applied on the geom and so assume that the texmatrix will be handled correctly (but it's risky), add a new shader attrib (F_shader_point_tex_matrix ?), or just add a warning to the user if the case arise.
